pkg/kvstore/redis.go

```diff
@@ -162,13 +169,50 @@ func (s *redisKvStore) PutBatch(ctx context.Context, values interface{}) error {
 		return fmt.Errorf("could not convert values from %T to map[interface{}]interface{}", values)
 	}
 
+	chunkSize := s.settings.BatchSize
+	pairs := make([]interface{}, 0, chunkSize)
 	for k, v := range mii {
-		if err = s.Put(ctx, k, v); err != nil {
-			return fmt.Errorf("failed to write batch to redis: %w", err)
+		key, value, err := s.MarshalKeyValue(k, v)
+		if err != nil {
+			return fmt.Errorf("PutBatch could not marshal key/value: %w", err)
+		}
+		pairs = append(pairs, key)
+		pairs = append(pairs, value)
+
+		if len(pairs) >= 2*chunkSize {
+			err = s.flushChunk(ctx, pairs)
+			if err != nil {
+				return fmt.Errorf("failed to write batch to redis: %w", err)
+			}
+			pairs = make([]interface{}, 0, chunkSize)
 		}
 	}
 
-	return nil
+	return s.flushChunk(ctx, pairs)
+}
+
+func (s *redisKvStore) flushChunk(ctx context.Context, pairs []interface{}) error {
+	if len(pairs) < 1 {
+		return nil
+	}
+
+	err := s.client.MSet(ctx, pairs)
+	if err != nil {
+		return fmt.Errorf("failed to flush batch to redis: %w", err)
+	}
+
+	// setting ttl
+	pipe := s.client.Pipeline()
+	for i := 0; i < len(pairs); i += 2 {
+		keyStr, ok := pairs[i].(string)
```

Comment:
This is code controlled completely by you, so this should never fail. Therefore I would not check the assertion, but simply panic if it is not the case.

Reply:
actually, i would prefer this assertion check instead of the panic..  
feels just more clean.. you're right, that this can fail only by changing some code, but if some other developer changes something somewhere, he would get a nice hint about what went wrong? :thinking:

Reply:
reverted to the assertion check.

Reply:
I would argue the hint would be the failing test the dev should then get. And at least for me it is much easier to debug a panic than an error. Why? Because I can set a breakpoint for a panic (I think it is somewhere in the runtime in fatalpanic or something like that for me) and then immediately get the state at the panic - with an error, I always have to trace that. But if you would like to keep the error, that is also fine with me, in the end it should be covered by the test anyway.